src/core/manager.c

```diff
@@ -732,16 +732,17 @@ int manager_new(UnitFileScope scope, unsigned test_run_flags, Manager **_m) {
                 goto fail;
         }
 
-        if (MANAGER_IS_SYSTEM(m)) {
-                r = mkdir_label("/run/systemd/units", 0755);
-                if (r < 0 && r != -EEXIST)
-                        goto fail;
-        }
-
-        /* Note that we do not set up the notify fd here. We do that after deserialization,
-         * since they might have gotten serialized across the reexec. */
+        if (test_run_flags == 0) {
+                if (MANAGER_IS_SYSTEM(m)) {
+                        r = mkdir_label("/run/systemd/units", 0755);
+                        if (r < 0 && r != -EEXIST)
+                                goto fail;
+                }
 
-        m->taint_usr = dir_is_empty("/usr") > 0;
+                /* Note that we do not set up the notify fd here. We do that after deserialization,
+                 * since they might have gotten serialized across the reexec. */
+                m->taint_usr = dir_is_empty("/usr") > 0;
```

Comment:
Why move this part under the conditional?

Reply:
testing whether /usr is empty is actually kinda pointless anyway at this time, because it will be mounted already at that time, and we override the field later. Given that `--test` is generally not called as PID 1 super-duper early on it's hence unnecessary to check this here.

That said, I think we should drop this here, the value is useless here as as mentioned we override it later anyway with the value we collected much earlier. I'll rework the PR to remove the test altogether

Reply:
hmm, seems I was wrong there. There's no /usr check earlier than this, and it's not necessary either. Anyway I dropped the extra inclusion in the branch now, but slightly tweaked the tain usr check to not bother with the initrd